platform/posix/transport/src/plaintext_posix.c

```diff
@@ -115,79 +115,40 @@ int32_t Plaintext_Recv( NetworkContext_t * pNetworkContext,
-                                   SO_RCVTIMEO,
-                                   &recvTimeout,
-                                   &recvTimeoutLen );
-
-    /* Make #select return immediately if getting the timeout failed. */
-    if( getTimeoutStatus < 0 )
-    {
-        recvTimeout.tv_sec = 0;
-        recvTimeout.tv_usec = 0;
-    }
 
-    /* MISRA Directive 4.6 flags the following line for a violation of using a
-     * basic type "int" rather than a type that includes size and signedness information.
-     * We suppress the violation as the flagged type, "fd_set", is a POSIX
-     * system-specific type, and is used for the call to "select()". */
-
-    /* MISRA Rule 14.4 flags the following line for using condition expression "0"
-     * as a boolean type. We suppress the violation as the "FD_ZERO" is a POSIX
-     * specific macro utility whose implementation is supplied by the system.
-     * The "FD_ZERO" macro is called as specified by the POSIX manual here:
-     * https://pubs.opengroup.org/onlinepubs/009695399/basedefs/sys/select.h.html */
-    /* coverity[misra_c_2012_directive_4_6_violation] */
-    /* coverity[misra_c_2012_rule_14_4_violation] */
-    FD_ZERO( &readfds );
-
-    /* MISRA Directive 4.6 flags the following line for a violation of using a
-     * basic type "int" rather than a type that includes size and signedness information.
-     * We suppress the violation as the flagged type, "fd_set", is a POSIX
-     * system-specific type, and is used for the call to "select()". */
-    /* coverity[misra_c_2012_directive_4_6_violation] */
-
-    /* MISRA Rule 10.1, Rule 10.8 and Rule 13.4 flag the following line for
-     * implementation of the "FD_SET()" POSIX macro. We suppress these violations
-     * "FD_SET" is a POSIX specific macro utility whose implementation
-     * is supplied by the system.
-     * The "FD_SET" macro is used as specified by the POSIX manual here:
-     * https://pubs.opengroup.org/onlinepubs/009695399/basedefs/sys/select.h.html */
-    /* coverity[misra_c_2012_directive_4_6_violation] */
-    /* coverity[misra_c_2012_rule_10_1_violation] */
-    /* coverity[misra_c_2012_rule_13_4_violation] */
-    /* coverity[misra_c_2012_rule_10_8_violation] */
-    FD_SET( pPlaintextParams->socketDescriptor, &readfds );
+    /* Initialize the file descriptor.
+     * #POLLPRI corresponds to high-priority data while #POLLIN corresponds
+     * to any other data that may be read. */
+    pollFds.events = POLLIN | POLLPRI;
+    pollFds.revents = 0;
+    /* Set the file descriptor for poll. */
+    pollFds.fd = pPlaintextParams->socketDescriptor;
 
-    /* Check if there is data to read from the socket. */
-    selectStatus = select( pPlaintextParams->socketDescriptor + 1,
-                           &readfds,
-                           NULL,
-                           NULL,
-                           &recvTimeout );
+    /* Speculative read for the start of a payload. */
```

Comment:
```suggestion
    /* Speculative read for the start of a payload. */
    /* Note: This is done to avoid blocking behavior of recv() call when no data is available from the network to read. */
```